### PR TITLE
Update streaming Readme /1

### PR DESCRIPTION
**Pull request by Rick-Anderson**
@logiclrd please review, this ReadMe needs some more work.
We don't need to give details on running in VS. BTW, Kestrel is launch by default.

I've tried all the samples and they just spin and nothing is played.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@logiclrd  the browser doesn't show output, `curl  http://localhost:5100/v1/sing --no-buffer` works.
How do I get output from the browser?

File: fundamentals/middleware/rate-limit/WebRate2/Program.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#define SECOND // FIRST SECOND
+#define FIRST // FIRST SECOND
```
Comment (logiclrd) [inline fundamentals/middleware/rate-limit/WebRate2/Program.cs:1]: Is it intentional for this to be part of this PR?

Comment (Rick-Anderson) [reply fundamentals/middleware/rate-limit/WebRate2/Program.cs:1]: Nope but it doesn't hurt anything

**Comment by logiclrd** [comment]
> How do I get output from the browser?

That is extremely strange. I can replicate the problems you're encountering, but I wasn't seeing this behaviour when I was last working on this. Investigating...

**Comment by logiclrd** [comment]
Okay I'm pretty confused right now. When I added this, it worked. I don't think I've changed anything, but now it doesn't work -- except for with `curl`. Why would `curl` be different??

I don't actually even need to specify `--no-buffer` with `curl`, though I do need to specify `--output -` because it doesn't want to dump raw binary data to the console by default (which is polite :-) ).

When I try to pass `curl` through Fiddler to find out how it's different, it doesn't get output and I see the process memory growing rapidly.

File: fundamentals/streaming/README.md [hunk 2]
```diff
@@ -4,12 +4,13 @@ This repository contains sample code for the article "Streaming Responses in ASP
...
 
-The project contains two implementations of the definition of a simple ASP.NET Core web site that contains endpoints that sing a song:
+The project contains three implementations of the definition of a basic ASP.NET Core web site that contains endpoints that play a song:
 
 * `AsynchronousWithSystemTextJson`
 * `SynchronousWithNewtonsoftJson`
+* `DynamicBinaryStream`
 
-These endpoints sing the song indefinitely, streaming the song repeatedly until the client disconnects. This acts as a demonstration of how to implement streaming endpoints with ASP.NET Core.
+These endpoints play the song indefinitely, streaming the song repeatedly until the client disconnects. This acts as a demonstration of how to implement streaming endpoints with ASP.NET Core.
```
Comment (logiclrd) [inline fundamentals/streaming/README.md:13]: I'm not sure this change makes sense. They literally do _not_ "play" the song.

**Comment by logiclrd** [comment]
The memory usage may a red herring, it stops growing after a bit. Also, even though nothing is showing up in the browser, I see a similar memory usage pattern.

**Comment by logiclrd** [comment]
The other two samples aren't working either. Something must have changed in the framework. :-/

**Comment by logiclrd** [comment]
I think it must be a change in the framework, because I get the same behaviour with Internet Explorer.

**Comment by logiclrd** [comment]
Okay I have narrowed it down. If the `Accept` header contains MIME type `text/html`, then _none_ of the streaming samples work.

Do you have any idea why this would be??

**Comment by logiclrd** [comment]
Further investigation: If I change the target framework from `net7` to `net6`, then it works.

**Comment by logiclrd** [comment]
Is there any possibility of finding someone with a more intimate knowledge/understanding of ASP.NET Core's internals, who might know why 6.0 works and 7.0 doesn't with `Accept: text/html`??

**Comment by Rick-Anderson** [comment]
> Okay I have narrowed it down. If the `Accept` header contains MIME type `text/html`, then _none_ of the streaming samples work.
> 




> Further investigation: If I change the target framework from `net7.0` to `net6.0`, then it works.

@Tratcher  can you help explain what's going on?

**Comment by Tratcher** [comment]
AsynchronousWithSystemTextJson doesn't work for me on net6.0 either. Everything seems to be getting buffered in memory, it never makes it to Kestrel. Which one were you testing on net6.0?

> Okay I have narrowed it down. If the Accept header contains MIME type text/html, then none of the streaming samples work.

This seems to be the cause, not the TFM.  @captainsafia who can take a look?

**Comment by logiclrd** [comment]
> AsynchronousWithSystemTextJson doesn't work for me on net6.0 either

That's odd, it works for me on net6.0 and on net7.0. On net6.0, it works in a Chrome tab.

image

I am unable to reproduce the problem you're seeing.

File: fundamentals/streaming/README.md (hunk 2 above)
Comment (Rick-Anderson) [reply fundamentals/streaming/README.md:13]:
```suggestion
These endpoints stream the song indefinitely, repeating until the client disconnects. This acts as a demonstration of how to implement streaming endpoints with ASP.NET Core.
```

**Comment by Rick-Anderson** [comment]
@logiclrd I'm going to merge this, thanks for all the help. It works in .NET 7 for sure, that's the main thing.

**Comment by Rick-Anderson** [comment]
@logiclrd  check out [is:open is:issue label:"code help wanted" ](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aopen+is%3Aissue+label%3A%22code+help+wanted%22) in the ASP.NET Core repo.

**Pull request merged**